domain/user/service/service.go

```diff
@@ -136,13 +162,33 @@ func ValidateUsername(name string) error {
 // - usererrors.AlreadyExists: If a user with the supplied name already exists.
 // - usererrors.UserCreatorNotFound: If a creator has been supplied for the user
 // and the creator does not exist.
-func (s *Service) AddUser(ctx context.Context, user user.User) error {
-	if err := ValidateUsername(user.Name); err != nil {
-		return fmt.Errorf("username %q: %w", user.Name, err)
+func (s *Service) AddUser(ctx context.Context, usr user.User, creatorUUID user.UUID) error {
+	// Validate user name and creator name
+	if err := ValidateUsername(usr.Name); err != nil {
+		return errors.Annotatef(err, "username %q", usr.Name)
+	}
+
+	// Validate creator UUID
+	if err := creatorUUID.Validate(); err != nil {
+		return errors.Annotatef(err, "creator uuid %q", creatorUUID)
+	}
+
+	// Ger creator user
+	creatorUser, err := s.GetUser(ctx, creatorUUID)
```

Comment:
You don't actually need to do this, because there is a self-referencing constraint on the user table that ensures a user with this creator UUID exists. Just validate and do the insertion. I believe we can just catch the FK violation error and return this specific no-found one.

Reply:
Agreed, this was a rudiment from the implementation, I thought removed all of them.

Thx for pointing.